### DISCUSSION TITLE
Get all possible agent locations

### DISCUSSION THREAD

**Discussion by roymatza** [allenai/ai2thor#865] [Q&A] [answered]
Hello,

I wonder if there's some way to get all the **valid** positions for the agent i.e., positions in the floor with no obstacles.
The current API only allows to get the reachable positions.

Thank you.

**Comment by mattdeitke** [comment, marked as answer]
Hi @roymatza, I would recommend:
1. Disable all objects from the scene (except the `Floor`).
2. Call `GetReachablePositions` to get the reachable positions without any objects.
3. Re-Enable all the objects in the scene.

Here is a notebook that reproduces that: https://colab.research.google.com/drive/1Y3SumyeWbWHrgJ7mHlkk6nEoVfBmgQ9U?usp=sharing.

Let me know if this works :)

**Reply by roymatza** [reply]
Works like a charm!
Thank you!
